event-publisher/src/main/java/uk/gov/ons/ctp/common/event/EventPublisher.java

```diff
@@ -118,46 +119,50 @@ public String sendEvent(
    * @return String UUID transaction Id for event
    */
   public String sendEvent(EventBackupData event) {
-    EventType type = event.getEventType();
+    TopicType type = event.getTopicType();
     SendInfo sendInfo = type.getBuilder().create(event.getEvent());
     if (sendInfo == null) {
       log.error("Unrecognised event type", kv("type", type));
       throw new UnsupportedOperationException("Unknown event: " + type);
     }
-    String transactionId = doSendEvent(type, sendInfo);
-    log.debug("Sent {} with transactionId {}", event.getEventType(), transactionId);
-    return transactionId;
+    GenericEvent genericEvent = doSendEvent(type, sendInfo);
+    String messageId = genericEvent.getHeader().getMessageId();
+    String correlationId = genericEvent.getHeader().getCorrelationId();
+
+    log.debug("Sent {} with messageId {}", event.getTopicType(), messageId);
+    log.debug("Sent {} with correlationId {}", event.getTopicType(), correlationId);
+    return messageId;
   }
 
-  private String doSendEvent(EventType eventType, SendInfo sendInfo) {
+  private GenericEvent doSendEvent(TopicType topicType, SendInfo sendInfo) {
     EventPayload payload = sendInfo.getPayload();
 
-    if (!payload.getClass().equals(eventType.getPayloadType())) {
+    if (!payload.getClass().equals(topicType.getPayloadType())) {
       log.error(
           "Payload incompatible for event type",
           kv("payloadType", payload.getClass()),
-          kv("eventType", eventType));
+          kv("eventType", topicType));
       String errorMessage =
           "Payload type '"
               + payload.getClass()
               + "' incompatible for event type '"
-              + eventType
+              + topicType
               + "'";
       throw new IllegalArgumentException(errorMessage);
     }
 
-    EventTopic eventTopic = EventTopic.forType(eventType);
+    EventTopic eventTopic = EventTopic.forType(topicType);
     if (eventTopic == null) {
-      log.error("Routing key for eventType not configured", kv("eventType", eventType));
-      String errorMessage = "Routing key for eventType '" + eventType + "' not configured";
+      log.error("Routing key for eventType not configured", kv("eventType", topicType));
+      String errorMessage = "Routing key for eventType '" + topicType + "' not configured";
```

Comment:
Shall we call a spade a spade?  ie "topicType"

Reply:
This has been updated.